### ISSUE TITLE
Axios rejections don't filter through into promise rejection for 503. 

### ISSUE THREAD

**Issue by rageycomma** [DiegoZoracKy/oas-client#2]
This is something I've noticed using oas-client when the service I'm pinging all of a sudden goes down, axios returns the 503, but oas-client doesn't filter that back through to the created client. 

```javascript
 const client = oasClient.create(schema); 
 client.doAction({ ... }).then((s) => {})
 .catch((e) => {}); // Never gets called on 503
```

Checking, it looks like the path `lib/make-api-call.js:makeApiCall` is the cause.

**Comment by DiegoZoracKy** [comment]
Hi @rageycomma, thanks for bringing this issue. I'll take a look at it, but if you would like to send a PR, feel free to do it.

**Comment by rageycomma** [comment]
Of course, if I get time I'll do a PR :) I just wanted to make you aware of it!

**Comment by DiegoZoracKy** [comment]
Great. Indeed, thanks for letting me know.
